runtime/src/impls.rs

```diff
@@ -289,10 +292,79 @@ impl Dispatchable for DispatchableSubstrateBridgeCall {
+            bridge_types::substrate::BridgeCall::SubstrateApp(_) => {
+                todo!("not inplemented")
+            }
+        }
+    }
+}
+
+#[cfg(feature = "wip")]
+impl Dispatchable for DispatchableSubstrateBridgeCall {
+    type RuntimeOrigin = crate::RuntimeOrigin;
+    type Config = crate::Runtime;
+    type Info = DispatchInfo;
+    type PostInfo = PostDispatchInfo;
+
+    fn dispatch(
+        self,
+        origin: Self::RuntimeOrigin,
+    ) -> sp_runtime::DispatchResultWithInfo<Self::PostInfo> {
+        frame_support::log::debug!("Dispatching SubstrateBridgeCall: {:?}", self.0);
+        match self.0 {
+            bridge_types::substrate::BridgeCall::ParachainApp(msg) => {
+                let call: parachain_bridge_app::Call<crate::Runtime> = msg.into();
+                let call: crate::RuntimeCall = call.into();
+                call.dispatch(origin)
+            }
+            bridge_types::substrate::BridgeCall::XCMApp(_msg) => Err(DispatchErrorWithPostInfo {
+                post_info: Default::default(),
+                error: DispatchError::Other("Unavailable"),
+            }),
+            bridge_types::substrate::BridgeCall::DataSigner(msg) => {
+                let call: bridge_data_signer::Call<crate::Runtime> = msg.into();
+                let call: crate::RuntimeCall = call.into();
+                call.dispatch(origin)
+            }
+            bridge_types::substrate::BridgeCall::MultisigVerifier(msg) => {
+                let call: multisig_verifier::Call<crate::Runtime> = msg.into();
+                let call: crate::RuntimeCall = call.into();
+                call.dispatch(origin)
+            }
+            bridge_types::substrate::BridgeCall::SubstrateApp(msg) => {
+                let call: substrate_bridge_app::Call<crate::Runtime> = msg.try_into()?;
+                let call: crate::RuntimeCall = call.into();
+                call.dispatch(origin)
+            }
+        }
+    }
+}
+
+#[cfg(not(feature = "wip"))]
+impl GetDispatchInfo for DispatchableSubstrateBridgeCall {
+    fn get_dispatch_info(&self) -> DispatchInfo {
+        match &self.0 {
+            bridge_types::substrate::BridgeCall::ParachainApp(msg) => {
+                let call: parachain_bridge_app::Call<crate::Runtime> = msg.clone().into();
+                call.get_dispatch_info()
+            }
+            bridge_types::substrate::BridgeCall::XCMApp(_msg) => Default::default(),
+            bridge_types::substrate::BridgeCall::DataSigner(msg) => {
+                let call: bridge_data_signer::Call<crate::Runtime> = msg.clone().into();
+                call.get_dispatch_info()
+            }
+            bridge_types::substrate::BridgeCall::MultisigVerifier(msg) => {
+                let call: multisig_verifier::Call<crate::Runtime> = msg.clone().into();
+                call.get_dispatch_info()
+            }
+            bridge_types::substrate::BridgeCall::SubstrateApp(_) => {
+                todo!("not implemented")
```

Comment:
We have `unimplemented!` macro for something that will be not implemented. In case of todo, it is strongly advised to leave a comment with a gh issue.